SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp

```diff
@@ -157,10 +161,71 @@ void USpatialReceiver::OnRemoveEntity(Worker_RemoveEntityOp& Op)
 
 void USpatialReceiver::OnAuthorityChange(Worker_AuthorityChangeOp& Op)
 {
-	if (Op.component_id == SpatialConstants::GLOBAL_STATE_MANAGER_COMPONENT_ID
-		&& Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	if (bInCriticalSection)
+	{
+		PendingAuthorityChanges.Add(Op);
+		return;
+	}
+
+	HandleActorAuthority(Op);
+}
+
+void USpatialReceiver::HandleActorAuthority(Worker_AuthorityChangeOp& Op)
+{
+	if (NetDriver->IsServer())
 	{
-		GlobalStateManager->ExecuteInitialSingletonActorReplication();
+		if (Op.component_id == SpatialConstants::GLOBAL_STATE_MANAGER_COMPONENT_ID
+			&& Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+		{
+			GlobalStateManager->ExecuteInitialSingletonActorReplication();
+			return;
+		}
+
+		// If we became authoritative over the position component. set our role to be ROLE_Authority
+		// and set our RemoteRole to be ROLE_AutonomousProxy iff the actor has an owning connection.
+		if (Op.component_id == SpatialConstants::POSITION_COMPONENT_ID)
+		{
+			if (AActor* Actor = NetDriver->GetEntityRegistry()->GetActorFromEntityId(Op.entity_id))
+			{
+				if (Op.authority == WORKER_AUTHORITY_AUTHORITATIVE)
```

Comment:
Not possible to hit this with authority loss imminent?

Reply:
We should do another pass when we want to properly enable soft handover.

Reply:
I created UNR-640 for you, please add as a comment here and the other place